lib/Doctrine/DBAL/Cache/ResultCacheStatement.php

```diff
@@ -336,13 +341,16 @@ private function saveToCache(): void
             return;
         }
 
-        $data = $this->resultCache->fetch($this->cacheKey);
-        if (! $data) {
-            $data = [];
+        if ($this->fetchedData === null) {
+            $this->fetchedData = $this->resultCache->fetch($this->cacheKey);
+        }
+
+        if (! $this->fetchedData) {
```

Comment:
Please avoid implicit conversion to boolean. It's not allowed in newer branches.

Reply:
Ok. This line was written before me - I've just change the variable name.

https://github.com/doctrine/dbal/blob/2.11.x/lib/Doctrine/DBAL/Cache/ResultCacheStatement.php#L340